### PR TITLE
Refine InstructorCourseDetailsPageAction Slice

### PR DESCRIPTION
**Pull request by bamboo3250**
refined Logic.java
refined StudentAttributes.java
refined StringHelper.java
refined AjaxResult.java
Refined InstructorCourseDetailsAction.java

Added a test case to fully cover all the branches in InstructorCourseDetailsAction.java

### CODE REVIEW

**Comment by wkurniawan07** [comment]
appreciate your hard work, but I've done StudentAttributes already :v:

**Comment by bamboo3250** [comment]
Oh I didn't see that. Luckily it's the last commit so I can reverse it. I'll do that tmr.

File: src/main/java/teammates/logic/api/Logic.java [hunk 1]
```diff
@@ -726,11 +719,11 @@ public CourseDetailsBundle getCourseDetails(String courseId)
      * @return A more detailed version of courses for this instructor. 
      *   Returns an empty list if none found.\
      */
-    public HashMap<String, CourseDetailsBundle> getCourseDetailsListForInstructor(
-            String instructorId, boolean omitArchived) throws EntityDoesNotExistException {
+    public HashMap<String, CourseDetailsBundle> getCourseDetailsListForInstructor(String instructorId, 
+                                                                                  boolean omitArchived) 
+            throws EntityDoesNotExistException {
```
Comment (damithc) [inline src/main/java/teammates/logic/api/Logic.java]: Looks like the original is slightly more readable.

File: src/main/java/teammates/logic/api/Logic.java [hunk 2]
```diff
@@ -908,22 +888,19 @@ private void ____STUDENT_level_methods__________________________________() {
...
         studentsLogic.createStudentCascade(student);
-
     }
 
-    public void createStudentWithoutDocument(StudentAttributes student)
-            throws EntityAlreadyExistsException, InvalidParametersException, EntityDoesNotExistException {
-        
+    public void createStudentWithoutDocument(StudentAttributes student) throws EntityAlreadyExistsException, 
+                                                                               InvalidParametersException, 
+                                                                               EntityDoesNotExistException {
```
Comment (damithc) [inline src/main/java/teammates/logic/api/Logic.java]: Original is better. It is as readable as the new version but uses fewer lines.

File: src/main/java/teammates/logic/api/Logic.java [hunk 3]
```diff
@@ -1167,8 +1119,8 @@ public void resetStudentGoogleId(String originalEmail, String courseId) throws I
      * Preconditions: <br>
      * * All parameters are non-null.
      */
-    public void updateStudent(String originalEmail, StudentAttributes student)
-            throws InvalidParametersException, EntityDoesNotExistException {
+    public void updateStudent(String originalEmail, StudentAttributes student) throws InvalidParametersException, 
+                                                                                      EntityDoesNotExistException {
```
Comment (damithc) [inline src/main/java/teammates/logic/api/Logic.java]: The revised version is ok but the original is acceptable too.

File: src/main/java/teammates/logic/api/Logic.java [hunk 4]
```diff
@@ -1383,11 +1331,10 @@ private void ____FEEDBACK_SESSION_level_methods_____________________________() {
      * Preconditions: <br>
      * * All parameters are non-null.
      */
-    public void createFeedbackSession(FeedbackSessionAttributes feedbackSession)
-            throws EntityAlreadyExistsException, InvalidParametersException, EntityDoesNotExistException {
-        
+    public void createFeedbackSession(FeedbackSessionAttributes feedbackSession) throws EntityAlreadyExistsException, 
+                                                                                        InvalidParametersException, 
+                                                                                        EntityDoesNotExistException {
```
Comment (damithc) [inline src/main/java/teammates/logic/api/Logic.java]: original is preferable.

File: src/main/java/teammates/logic/api/Logic.java [hunk 5]
```diff
@@ -1787,7 +1743,10 @@ public FeedbackQuestionAttributes createFeedbackQuestionForTemplate(
      * Preconditions: <br>
      * * All parameters are non-null.
      */
-    public FeedbackQuestionAttributes copyFeedbackQuestion(String feedbackQuestionId, String feedbackSessionName, String courseId, String instructorEmail)
+    public FeedbackQuestionAttributes copyFeedbackQuestion(String feedbackQuestionId, 
+                                                           String feedbackSessionName, 
+                                                           String courseId, 
+                                                           String instructorEmail)
```
Comment (damithc) [inline src/main/java/teammates/logic/api/Logic.java]: When you have many parameters, consider having multiple parameters in the same line. In this case, we can have 2 per line, saving two lines.

**Comment by damithc** [comment]
Added a few minor comments about style. After fixing them, can pass to Hong Jin for final review.

**Comment by bamboo3250** [comment]
revised the coding style.

@kanghj later please review this one when u have time.

**Pull request merged**